noir/compiler/noirc_frontend/src/node_interner.rs

```diff
@@ -1042,6 +1043,34 @@ impl NodeInterner {
         Ok(impl_kind)
     }
 
+    /// Given a `ObjectType: TraitId` pair, find all implementations without taking constraints into account or
+    /// applying any type bindings. Useful to look for a specific trait in a type that is used in a macro.
+    pub fn lookup_all_trait_implementations(
+        &self,
+        object_type: Type,
```

Comment:
This can be a reference and it'd remove some cloning in the aztec_macros side I think

Reply:
Referenced, but realistically didn't make much of a difference